apps/dapp-console-api/src/routes/contracts/ContractsRoute.ts

```diff
@@ -48,6 +52,95 @@ export class ContractsRoute extends Route {
       }
     })
 
+  public readonly createContract = 'createContract' as const
+  public readonly createContractController = this.trpc.procedure
+    .use(isPrivyAuthed(this.trpc))
+    .input(
+      this.z.object({
+        contractAddress: zodEthereumAddress,
+        deploymentTxHash: zodEthereumTransaction,
+        deployerAddress: zodEthereumAddress,
+        chainId: this.z.number(),
+        appId: this.z.string(),
+      }),
+    )
+    .mutation(async ({ ctx, input }) => {
+      const { user } = ctx.session
+      const deploymentTxHash = input.deploymentTxHash as Hash
+      const contractAddress = input.contractAddress as Address
+      const deployerAddress = input.deployerAddress as Address
+      const { chainId, appId } = input
+
+      if (!user) {
+        throw Trpc.handleStatus(401, 'user not authenticated')
+      }
+
+      const publicClient = supportedChainsPublicClientsMap[chainId]
+
+      if (!publicClient) {
+        throw Trpc.handleStatus(400, 'chain not supported')
+      }
+
+      const deploymentTx = await publicClient
+        .getTransactionReceipt({
+          hash: deploymentTxHash,
+        })
+        .catch((err) => {
+          metrics.listContractsErrorCount.inc({ chainId })
+          this.logger?.error(
+            {
+              error: err,
+              entityId: ctx.session.user?.entityId,
+              txHash: deploymentTxHash,
+              chainId,
+            },
+            'error fetching tx',
+          )
+          throw Trpc.handleStatus(500, 'error fetching deployment transaction')
```

Comment:
nit: this can prob be a 400 error since the user supplied a tx hash that isn't valid